skimage/feature/corner.py

```diff
@@ -968,8 +968,8 @@ def corner_peaks(image, min_distance=1, threshold_abs=None, threshold_rel=0.1,
         coords = np.transpose(peaks.nonzero())
         for r, c in coords:
             if peaks[r, c]:
```

Comment:
I don't think that this check is necessary since `r, c` comes from `peaks.nonzero()`?

Reply:
I guess the problem is in how we are doing subtraction here. I've been printing out the index locations corner_peaks is trying to set. Let's say peak_local_max returns two connected peaks to corner_peaks at [0:1, 0:2], and we have min_distance=2 for example.
 
The returned values from "coords = np.transpose(peaks.nonzero())":
 [[0 0]
 [0 1]]

Now, we get this: 
`Rows = [0-2: 0+2+1],  Cols = [0-2: 0+2+1]`

And now we try to do this:
`img[-2:3, -2:3] = false`

It's not able to set those locations to false. We should end up with one peak here and instead we get two.

![test4](https://user-images.githubusercontent.com/16522814/66316368-8d763580-e8e5-11e9-9fb2-27fc9cd0e250.png)


To Reproduce:
```
import numpy as np
import matplotlib.pyplot as plt
from skimage.feature import corner_peaks


img = np.zeros((5, 5), dtype=np.float32)
img[0:1, 0:2] = 1

corners = corner_peaks(img, exclude_border=False, min_distance=2)

plt.imshow(img, cmap='gray')
plt.plot(corners[:, 1], corners[:, 0], 'bo', markersize=2)
plt.show()
```

Reply:
Ah, you're right! Sorry :-p

Reply:
May we then use `scipy.spatial.distance.pdist` to find the points that are too close to each others instead of modifying the `peaks` array iteratively?

Reply:
@rfezzani No worries, thanks for reviewing! I like the idea. It's interesting how it actually goes about updating the array. 

Is there a performance benefit to using `scipy.spatial.distance.pdist` for this? We need to guarantee we are grabbing the top left corner within the min_distance region. So, I believe we'd still have to iterate through the entire peaks array, but this time do additional comparisons to min_distance. My thought is that we would still be modifying the array, only this time more selectively?

Reply:
Hi @Tea7ime, sorry for my late answer. I played a bit with the `pdist` function.

```python
def corner_peaks_new(image, min_distance=1, threshold_abs=None,
                     threshold_rel=0.1, exclude_border=True, indices=True,
                     num_peaks=np.inf, footprint=None, labels=None):

    peaks = peak_local_max(image, min_distance=min_distance,
                           threshold_abs=threshold_abs,
                           threshold_rel=threshold_rel,
                           exclude_border=exclude_border,
                           indices=False, num_peaks=num_peaks,
                           footprint=footprint, labels=labels)
    if min_distance > 0:
        coords = np.transpose(peaks.nonzero())
        num_peaks = coords.shape[0]
        # Compute the distance between each detected distance
        dist_mat = pdist(coords, 'chebyshev')
        rejected_peaks = set()
        start = end = 0
        step = num_peaks - 1
        for i in range(num_peaks - 2):
            end += step
            if i in rejected_peaks:
                start = end
                step -= 1
                continue
            rejected_peaks.update(i + 1 + np.nonzero(dist_mat[start:end] <
                                                     min_distance)[0])
            start = end
            step -= 1

        # Remove the peaks that are too close to each others
        rejected_peaks = list(rejected_peaks)
        if indices is True:
            return np.delete(coords, rejected_peaks, axis=0)

        peaks[coords[rejected_peaks, :]] = False

    if indices is True:
        return np.transpose(peaks.nonzero())

    return peaks
```

This implementation is slower then the actual one (but still fast), but it allows to use different distances: the actual implementation corresponds to the Chebyshev distance (it considers a square area around the peaks), but `pdist` allows to use the Euclidean distance (a circle area around the peaks).

Reply:
I know that it is actually not the main concern of this PR. I will fill a new one instead of adding noise to this one... Sorry